projects/ebv-analyzer-app/src/app/ebv-selector/ebv-selector.component.ts

```diff
@@ -350,335 +243,51 @@ export class EbvSelectorComponent implements OnInit, OnDestroy {
-                },
-            },
-        };
-
-        let colorizer = Colorizer.fromDict(this.ebvTree.tree.colorizer);
-
-        const currentGroup = this.ebvPath[this.ebvPath.length - 1];
-        if (currentGroup.dataRange) {
-            // rescale colors in colorizer
-
-            if (colorizer instanceof LinearGradient) {
-                const step = (currentGroup.dataRange[1] - currentGroup.dataRange[0]) / (colorizer.breakpoints.length - 1);
-
-                const breakpoints = [];
-
-                let value = currentGroup.dataRange[0];
-
-                for (const breakpoint of colorizer.breakpoints) {
-                    breakpoints.push(breakpoint.cloneWithValue(value));
-
-                    value += step;
-                }
-
-                colorizer = new LinearGradient(breakpoints, colorizer.noDataColor, colorizer.defaultColor);
-            } else if (colorizer instanceof LogarithmicGradient && currentGroup.dataRange[0] > 0) {
-                const step = (currentGroup.dataRange[1] - currentGroup.dataRange[0]) / (colorizer.breakpoints.length - 1);
-
-                const breakpoints = [];
-
-                let value = currentGroup.dataRange[0];
-
-                for (const breakpoint of colorizer.breakpoints) {
-                    breakpoints.push(breakpoint.cloneWithValue(value));
-
-                    value += step;
-                }
-
-                colorizer = new LogarithmicGradient(breakpoints, colorizer.noDataColor, colorizer.defaultColor);
-            }
-        }
-
-        return this.projectService.registerWorkflow(workflow).pipe(
-            map((workflowId) => {
-                const rasterLayer = new RasterLayer({
-                    name: 'EBV',
-                    workflowId,
-                    isVisible: true,
-                    isLegendVisible: false,
-                    symbology: new RasterSymbology(1.0, colorizer),
-                });
-
-                return rasterLayer;
-            }),
-        );
-    }
-
-    private request<T>(request: string, parameters: string | undefined, dataCallback: (data: T) => void): void {
-        this.loading$.next(true);
-
-        const parametersOpt = parameters ? `?${parameters}` : '';
-
-        this.userService
-            .getSessionTokenForRequest()
-            .pipe(
-                mergeMap((sessionToken) =>
-                    this.http.get<T>(`${this.config.API_URL}/${request}${parametersOpt}`, {
-                        headers: new HttpHeaders().set('Authorization', `Bearer ${sessionToken}`),
-                    }),
-                ),
-            )
-            .subscribe((data) => {
-                dataCallback(data);
-
+                const dataset = response.data[0];
+                this.preselectedPath = [
+                    dataset.ebv.ebv_class,
+                    dataset.ebv.ebv_name,
+                    dataset.title,
+                    0 /*default metric*/,
+                    0 /*default entity*/,
```

Comment:
Will this work if there is also a scenario?

Reply:
a8c0f6232f9c1b1e4d710ad1955cdab1d9789911